### PR TITLE
fix(theming): also apply enforced theme for guests

### LINKED ISSUE #41048: Theming: Force Light or Dark Theme Regardless of System
**Is your feature request related to a problem? Please describe.**
Uniform design of our Nextcloud web interface is important to us. Unfortunately, Nextcloud consistently looks entirely different depending on the device settings, especially for unregistered users with whom we want to share data.

**Describe the solution you'd like**
We desire a switch in the administration settings of Nextcloud that enforces either the light or dark theme of Nextcloud for both unauthenticated and authenticated users, regardless of device settings.

**Describe alternatives you've considered**
No alternative ideas. The default, current behavior of Nextcloud should not be influenced. For an alternative solution, such as using custom CSS rules, I would also be grateful.

**Additional context**
Nothing to add

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix https://github.com/nextcloud/server/issues/41048

### CODE REVIEW

**Comment by Oriann** [comment]
Any chance this will be backported to NC version 28 and 29 ?

File: apps/theming/lib/Service/ThemesService.php [hunk 1]
```diff
@@ -26,7 +26,7 @@ class ThemesService {
 
 	public function __construct(IUserSession $userSession,
 		IConfig $config,
```
Comment (susnux) [inline apps/theming/lib/Service/ThemesService.php:28]: Maybe then also refactor them if you do it for the default theme :)

```suggestion
	public function __construct(
		private IUserSession $userSession,
		private IConfig $config,
```

File: apps/theming/lib/Service/ThemesService.php [hunk 2]
```diff
@@ -52,6 +52,17 @@ public function __construct(IUserSession $userSession,
 	 * @return ITheme[]
 	 */
 	public function getThemes(): array {
+		// Enforced theme if configured
+		$enforcedTheme = $this->config->getSystemValueString('enforce_theme', '');
+		if ($enforcedTheme !== '' && isset($this->themesProviders[$enforcedTheme])) {
```
Comment (susnux) [inline apps/theming/lib/Service/ThemesService.php:57]: If the `isset` fails it should at least log a warning that the config is invalid for admins to figure out whats wrong.

**Review by AndyScherzinger — APPROVED** [review]

**Review by susnux — APPROVED** [review]

**Comment by Oriann** [comment]
PR broken in NC version 29.0.7

https://github.com/nextcloud/server/issues/41048#issuecomment-2385236674

**Pull request merged**